src/hooks/v1/ContractLoader.ts

```diff
@@ -54,8 +54,22 @@ const loadContract = async (
   network: NetworkName,
   signerOrProvider: JsonRpcSigner | JsonRpcProvider,
 ): Promise<Contract> => {
-  let contract = await import(
-    `@jbx-protocol/contracts-v1/deployments/${network}/${contractName}.json`
-  )
+  /**
+   * Conditionally load contracts based on configured network.
+   * This prevents ABIs for unused networks being imported to our JS bundle.
+   */
+  const _loadContract = async () => {
+    if (network === NetworkName.mainnet) {
+      return await import(
```

Comment:
something not working, sec.

Reply:
alternatively if there's any trouble with this, we could try something like:

```
import * as mainnetV1Contracts from '@jbx-protocol/contracts-v1/mainnet/...'
import * as rinkebyV1Contracts from '@jbx-protocol/contracts-v1/rinkeby/...'
...
if (process.env.REACT_APP_INFURA_NETWORK === 'mainnet') {
  return mainnetV1Contracts
}
```

Reply:
Yeah unfortunately we can't quite do that because that directory doens't have an `index.js` file

<img width="516" alt="Screen Shot 2022-02-12 at 12 45 04 pm" src="https://user-images.githubusercontent.com/94939382/153694632-128f6846-4e13-4f81-80f8-c83838a7956d.png">

The boring fix would be to import each `.json` file explicitly 🤷

Reply:
if we need to, we could request some architecture tweaks to the contracts package. though if your original method is working then no worries

Reply:
@peripheralist Bundle size with this change: 6.1MB => 4.15MB 😁 

Weirdly I don't see a difference in the total transferred size in development...not sure why.

<img width="1440" alt="Screen Shot 2022-02-12 at 12 48 13 pm" src="https://user-images.githubusercontent.com/94939382/153694744-996fd0bb-16a7-466b-88f5-d8dcf98a6e8b.png">

Reply:
> @peripheralist Bundle size with this change: 6.1MB => 4.15MB 😁
> 
> Weirdly I don't see a difference in the total transferred size in development...not sure why.
> 
> <img alt="Screen Shot 2022-02-12 at 12 48 13 pm" width="1440" src="https://user-images.githubusercontent.com/94939382/153694744-996fd0bb-16a7-466b-88f5-d8dcf98a6e8b.png">

sweet! where are you reading the 4.15?

Reply:
@aeolianeth sweet! where are you reading the 4.15 value?

Reply:
@peripheralist wow...sorry, it's 16.1 => 14.15. Was missing the `1` 😅

Reply:
hahaha it's a thicc app